src/java/org/apache/cassandra/service/accord/AccordCommandStore.java

```diff
@@ -251,15 +284,51 @@ public void processBlocking(Runnable runnable)
     }
 
     @Override
-    public <T> Future<T> submit(PreLoadContext loadCtx, Function<? super CommandStore, T> function)
+    public <T> Future<T> submit(PreLoadContext loadCtx, Function<? super SafeCommandStore, T> function)
     {
         AsyncOperation<T> operation = AsyncOperation.create(this, loadCtx, function);
         executor.execute(operation);
         return operation;
     }
 
     @Override
-    public Future<Void> execute(PreLoadContext preLoadContext, Consumer<? super CommandStore> consumer)
+    public Agent agent()
+    {
+        return agent;
+    }
+
+    @Override
+    public ProgressLog progressLog()
+    {
+        return progressLog;
+    }
+
+    @Override
+    public RangesForEpoch ranges()
+    {
+        return rangesForEpoch;
+    }
+
+    @Override
+    public long latestEpoch()
+    {
+        return time.epoch();
+    }
+
+    @Override
+    public Timestamp maxConflict(Keys keys)
```

Comment:
That's only because we haven't made it any more efficient here. This _should_ be implementation defined, so a bit of minor duplication for now seems fine.